DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/policer/test_policer_basic_functionality.py

```diff
@@ -0,0 +1,166 @@
+    2. Set link up on interfaces on all participant ports. Enslave all participant ports to the bridge.
+    3. Create an ingress qdisc on first port connected to Ixia port/shared block and add a rule with random selectors
+    4. Prepare a matching stream for the randomly selected selectors and with random generated size of packet.
+    5. Send traffic
+    6. Verify RX rate on RX ports is as expected (rate is limited by the police pass rule).
+    """
+
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 2)
+    if not tgen_dev or not dent_devices:
+        pytest.skip(
+            'The testbed does not have enough dent with tgen connections')
+    device = dent_devices[0]
+    dent = device.host_name
+    tg_ports = tgen_dev.links_dict[dent][0]
+    ports = tgen_dev.links_dict[dent][1][:2]
+    port_with_rule = ports[:1] if qdisc_type == 'port' else ports[1:]
+    bridge = 'bridge0'
+    block = random.randint(5, 3000)
+    frame_rate = 300000  # bps
+    tolerance = 0.12
+    # 1. Create a bridge entity and set link up on it.
+    out = await IpLink.add(input_data=[{dent: [{
+        'dev': bridge,
+        'type': 'bridge'}]
+    }])
+    assert out[0][dent]['rc'] == 0, 'Failed creating bridge.'
+
+    await IpLink.set(input_data=[{dent: [{'device': bridge, 'operstate': 'up'}]}])
+    assert out[0][dent]['rc'] == 0, 'Failed setting bridge to state UP.'
+
+    # 2. Set link up on interfaces on all participant ports. Enslave all participant ports to the bridge.
+    out = await IpLink.set(input_data=[{dent: [{
+        'device': port,
+        'operstate': 'up',
+        'master': bridge
+    } for port in ports]}])
+    assert out[0][dent]['rc'] == 0, 'Failed setting link to state UP.'
+
+    # 3. Create an ingress qdisc on first port connected to Ixia port/shared block and add a rule with random selectors
+    config = [{'dev': port, 'ingress_block': block, 'direction': 'ingress'} for port in port_with_rule]
+    if qdisc_type == 'port':
+        for item in config:
+            del item['ingress_block']
+
+    out = await TcQdisc.add(input_data=[{dent: config}])
+
+    rule_config = {
+        'action': {'police': {'rate': frame_rate, 'burst': frame_rate + 1000, 'conform-exceed': 'drop'}},
+        'want_ip': False if traffic_type == 'l2' else True,
+        'want_port': True if traffic_type in ['tcp', 'udp'] else None,
+        'skip_sw': True,
+        'want_mac': True if traffic_type == 'l2' else False,
+        'want_tcp': True if traffic_type == 'tcp' else False
+    }
+
+    tc_rule = tcutil_generate_rule_with_random_selectors(port_with_rule, **rule_config)
+
+    # Update tc rule protocol based on traffic type
+    if traffic_type == 'udp':
+        tc_rule['ip_proto'] = 'udp'
+    if traffic_type == 'ipv4':
+        tc_rule['protocol'] = 'ipv4'
+    if traffic_type == 'l2':
+        tc_rule['protocol'] = '0x8100'
+
+    if qdisc_type == 'port':
+        for port in port_with_rule:
```

Comment:
why use for loop? you can use a list comprehension:
```python
TcFilter.add(input_data=[{dent: [
    {**tc_rule, 'dev': port}
    for port in port_with_rule
]}])
```

Moreover if `port_with_rule` only ever contains a single port this can be simplified